cmake/SpectreParseTests.py

```diff
@@ -101,8 +101,18 @@ def parse_source_file(file_name):
         # "Unit.My.Test", "[Unit][My]"
         # group(1) == Unit.My.Test
         # group(2) == [Unit][My]
-        parsed_name = re.search(r'"(.*)",[\s]*"(.*)"', test_name)
-        test_name = parsed_name.group(1)
+        #
+        # Also handle cases where the name is very long and goes onto multiple
+        # lines (this requires the name to have parentheses because of how C++
+        # macros work),
```

Comment:
I don't understand this comment.  That's not something I would expect the preprocessor to care about.

Reply:
This is the use case I found that didn't work (on a branch, not develop):

```cpp
SPECTRE_TEST_CASE(("Unit.ParallelAlgorithms.Interpolation.Runtime.Callbacks."
                   "ObserveTimeSeriesOnSurface"),
                  "[Unit]") {
  test_option_parsing();
}
```

When I didn't have the `()` surrounding the test name, CMake complained. Also in other instances of macros, I found that I sometimes had to wrap a multi-line argument in `()`

Reply:
If CMake is complaining that's not a problem with macros.  CMake isn't running the preprocessor.

Have you checked that your test actually runs?  If I try something like what you have the test is silently ignored.

Reply:
Yup my test runs

Reply:
Did you re-run cmake?  If I copy-paste your example into a file it's silently ignored.

Reply:
Ok now it works with the test case

```cpp
SPECTRE_TEST_CASE("Unit.ParallelAlgorithms.Interpolation.Runtime.Callbacks."
                  "ObserveTimeSeriesOnSurface",
                  "[Unit]") {
  test_option_parsing();
}
```

I wasn't actually running the test through `ctest`. I was just running the executable directly and the test did run which is why I thought it had worked. Now I checked with `ctest`

Reply:
I think the regex I wrote in CMake is probably the part that's failing (or maybe it's regex in python, I forget what parses these nowadays). Just an FYI

Reply:
Turned out to be both :)